include/exadg/incompressible_navier_stokes/spatial_discretization/operator_dual_splitting.cpp

```diff
@@ -613,15 +609,8 @@ OperatorDualSplitting<dim, Number>::local_interpolate_velocity_dirichlet_bc_boun
     if(boundary_type == BoundaryTypeU::Dirichlet or boundary_type == BoundaryTypeU::DirichletCached)
     {
       integrator.reinit(face);
-      integrator.read_dof_values(dst);
-
```

Comment:
I had another look at this code and correctness issue I talked about. In the old code, this line (now deleted) made sure that we would keep the old DoF values set on other faces, as we only wrote the values for those entries located on the face via `face_to_cell_index_nodal`. But once we start interpolating from the face quadrature points to the cell, we lose the old values, so the proposed change will indeed run into the trap I mentioned in the other comment. I suggest to discuss this more generally, ideally by going to `VectorTools::interpolate_boundary_values`. As a side note, that function got considerably more efficient in the past few years when we implemented better interpolators for `MappingQ` via the infrastructure also used by `FEPointEvaluation`, and we can certainly have another look if necessary.